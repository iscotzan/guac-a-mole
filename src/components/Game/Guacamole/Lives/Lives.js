import React from "react";

const Lives = props => {
  const { visualRepresentation, lives } = props;
  let livesDisplay = [];
  if (typeof visualRepresentation === "number") {
    return (
      <div className="livesContainer">
        <h1>{lives}</h1>
      </div>
    );
  } else {
    for (let i = 0; i < lives; i++) {
      livesDisplay.push(
        <span className="singleLife" key={i}>
          {visualRepresentation}
        </span>
      );
    }
    return <div className="livesContainer">{livesDisplay}</div>;
  }
};

export default Lives;
