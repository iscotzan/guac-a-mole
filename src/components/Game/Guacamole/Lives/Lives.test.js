import React from "react";
import Lives from "./Lives";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<Lives visualRepresentation={'*'} lives={3} />);
describe("<Lives />", () => {
  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("renders without crashing", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
