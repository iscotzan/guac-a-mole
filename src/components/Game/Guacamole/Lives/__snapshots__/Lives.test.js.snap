// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lives /> renders without crashing 1`] = `
<div
  className="livesContainer"
>
  <span
    className="singleLife"
    key="0"
  >
    *
  </span>
  <span
    className="singleLife"
    key="1"
  >
    *
  </span>
  <span
    className="singleLife"
    key="2"
  >
    *
  </span>
</div>
`;
